app/controllers/authorships_controller.rb

```diff
@@ -221,9 +221,17 @@ def get_publication_via_sciencewire!(sw_id)
   # @return [Publication]
   def get_publication_via_wos!(author, wos_uid)
     WebOfScience.harvester.author_uid(author, wos_uid)
-    pub = Publication.find_by(wos_uid: wos_uid)
+    wossr = WebOfScienceSourceRecord.find_by(uid: wos_uid)
+    unless wossr
+      log_and_error!("A WebOfScienceSourceRecord was not found for WOS_UID:#{wos_uid}.")
+      return false
+    end
+
+    # we find the publication by looking for any matching publication for this WOS source record (could be MEDLINE or WOS ID, use all valid identifiers)
+    pub = wossr.record.matching_publication
+
     unless pub
-      log_and_error!("The publication with WOS_UID:#{wos_uid} was not found either locally or at WebOfScience.")
+      log_and_error!("A matching publication record for WOS_UID:#{wos_uid} was not found in the publication table.")
```

Comment:
Can this case get a test?
I how about doing `return pub if pub`. I'm not sure why https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/GuardClause isn't being enforced here.

Reply:
adjusted style and added a test case for the pub can't be found (and adjusted the other test case to catch the other specific error)